### ISSUE TITLE
Can't seem to get connected even with updated code

### ISSUE THREAD

**Issue by nutterb** [nutterb/redcapAPI#58]
```r
devtools::install_github("nutterb/redcapAPI")
library(redcapAPI)
rcon <- redcapConnection(url = 'https://www.redcapplugins.org/api/',
                         token = '[token]')
exportRecords(rcon)
```

Gives me the error

```
Error in curl::curl_fetch_memory(url, handle = handle) : 
  SSL connect error
```

Though I'm fairly certain this is a problem with my ability to connect, not with the code base itself.  Is that correct?

**Comment by wibeasley** [comment]
I agree.  I'm sorry, I didn't mean to imply that #57 would fix the SSL issue on the Vandy server.

**Comment by nutterb** [comment]
I think I understood that at some level, but wanted to be sure.  It looks like you just solved a problem I didn't know I had. :)  

This SSL error is killing me though.  I can get through on Postman.  I'm going to start trying to rebuild these on `RCurl` and see if I can get through.  I'm beginning to think this is a `curl` thing.

**Comment by wibeasley** [comment]
I'm betting it's the underlying [curl](http://curl.haxx.se/docs/caextract.html) that's the problem, instead of the [curl](https://github.com/jeroenooms/curl) R package.  That's one reason I was hoping @aarenson's PhpCap library would work (ie https://github.com/aarenson/PhpCap/issues/7).  And since it's working on other servers, I'm guessing Vandy's SSL is set up differently.  But this is really outside my expertise.

I tried, RCurl again without success.  But by all means I'd love to be wrong.  My attempt is at https://github.com/OuhscBbmc/REDCapR/blob/8e22b8d38f16df281a213f442e822591b652bace/playgrounds/ssl_playground.R#L71-L86.

The httr cert file is fairly old (Sept 2014); [it's actually bundled in R by default](https://github.com/hadley/httr/blob/e8a85c0a137c543090b60c694ea4d332e4833d68/R/utils.r#L72-L87).  I tried & [failed with a April 2015 cert file](https://github.com/OuhscBbmc/REDCapR/blob/8e22b8d38f16df281a213f442e822591b652bace/playgrounds/ssl_playground.R#L39).

**Comment by SteveLane** [comment]
The issue with redcapplugins could definitely be a curl issue. I tried to connect to the redcapplugins server, but got an error:
```
$ curl -v "https://redcapplugins.org/"
* About to connect() to redcapplugins.org port 443 (#0)
*   Trying 54.208.178.133...
* Connected to redcapplugins.org (54.208.178.133) port 443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
* NSS error -12286 (SSL_ERROR_NO_CYPHER_OVERLAP)
* Cannot communicate securely with peer: no common encryption algorithm(s).
* Error in TLS handshake, trying SSLv3...
> GET / HTTP/1.1
> User-Agent: curl/7.29.0
> Host: redcapplugins.org
> Accept: */*
>
* Connection died, retrying a fresh connect
* Closing connection 0
* Issue another request to this URL: 'https://redcapplugins.org/'
* About to connect() to redcapplugins.org port 443 (#1)
*   Trying 54.208.178.133...
* Connected to redcapplugins.org (54.208.178.133) port 443 (#1)
* TLS disabled due to previous handshake failure
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
* NSS error -12286 (SSL_ERROR_NO_CYPHER_OVERLAP)
* Cannot communicate securely with peer: no common encryption algorithm(s).
* Closing connection 1
curl: (35) Cannot communicate securely with peer: no common encryption algorithm(s).
```
This is from CentOS 7:
```
$ cat /etc/redhat-release arch
CentOS Linux release 7.0.1406 (Core)
```
with curl version:
```
$ curl -V
curl 7.29.0 (x86_64-redhat-linux-gnu) libcurl/7.29.0 NSS/3.15.4 zlib/1.2.7 libidn/1.28 libssh2/1.4.3
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp scp sftp smtp smtps telnet tftp
Features: AsynchDNS GSS-Negotiate IDN IPv6 Largefile NTLM NTLM_WB SSL libz
```
Found a similar issue in relation to CloudFare: http://www.webhostingtalk.com/showthread.php?t=1421536, so perhaps a curl upgrade may fix it, as it will include more encryption algs? I haven't yet had time to test.

**Comment by wibeasley** [comment]
Thanks for checking and researching, @SteveLane.  I've requested help updating the libraries on the test server.

I'm really hoping that these updates solve it, and my hope is strengthened after reading this in your linked CloudFare issue:
> It looks like this problem relates to ... NSS being out of date.

I've been able to get cURL working on some clients where it wasn't before.  But it's not a straight-forward update, and I don't hink it's even directly related to the cURL version number (sometimes older versions work on machines, compared to newer versions on others).  

I fear messing with the underlying cURL versions would be a nightmare.  I can't imagine making sure all the REDCapR & redcapAPI users link the correct libraries and bla bla bla.

So I'm really hoping updating the test server works.  Anyway, all these clients are working on the Oklahoma, Indiana, and Victoria servers, right?  It's really just this one EC2 test server.

I'll tell you when that gets updated.  If I can't get it working then, hopefully you'll see the problem.    Otherwise I might start tell people to export to to Excel.
